From 62e5823614732e8c1966a5e3dbda2b21b2d4d8a4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Sat, 4 Jun 2016 19:13:31 +0200
Subject: [PATCH] Add xen.cfg options for /mapbs and /noexitboot
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Organization: Invisible Things Lab
Cc: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>

On some platforms setting EFI command line parameters is unreliable.

Signed-off-by: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>
---
 xen/common/efi/boot.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/xen/common/efi/boot.c b/xen/common/efi/boot.c
index 125c9ce..ca23edc 100644
--- a/xen/common/efi/boot.c
+++ b/xen/common/efi/boot.c
@@ -1099,6 +1099,18 @@ efi_start(EFI_HANDLE ImageHandle, EFI_SYSTEM_TABLE *SystemTable)
         name.s = get_value(&cfg, section.s, "options");
         efi_arch_handle_cmdline(argc ? *argv : NULL, options, name.s);
 
+        name.s = get_value(&cfg, section.s, "mapbs");
+        if ( name.s )
+        {
+            map_bs = name.s[0] == '1';
+        }
+
+        name.s = get_value(&cfg, section.s, "noexitboot");
+        if ( name.s )
+        {
+            exit_boot_services = name.s[0] == '0';
+        }
+
         if ( !base_video )
         {
             name.cs = get_value(&cfg, section.s, "video");
-- 
2.5.5

